src/main/scala/edu/cmu/cs/obsidian/util/IdentityAstTransformer.scala

```diff
@@ -148,14 +148,14 @@ class IdentityAstTransformer {
             case s: StringLiteral => (s, List())
             case t: TrueLiteral => (TrueLiteral().setLoc(t), List())
             case f: FalseLiteral => (FalseLiteral().setLoc(f), List())
-            case t: This => (This().setLoc(t), List())
+            case t: This => (This(None).setLoc(t), List()) // todo: no idea if this is right
```

Comment:
For the identity transformer, just use whatever  type is already there (not None).

Reply:
got it!